.github/workflows/workflow.yml

```diff
@@ -216,21 +216,91 @@ jobs:
-          strip starship
-          tar czvf ../../../${{ matrix.name }} starship
-          cd -
+          strip target/${{ matrix.target }}/release/starship
+          tar czvf ${{ matrix.name }} target/${{ matrix.target }}/release/starship
+          echo "$(openssl dgst -sha256 ${{ matrix.name }} | awk '{print $2}') ${{ matrix.name }}" > ${{ matrix.name }}.sha256
+
+      - name: Upload build artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.name }}
+          path: ${{ matrix.name }}
+
+      - name: Upload shasum artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.name }}.sha256
+          path: ${{ matrix.name }}.sha256
+
+  # Create GitHub release with Rust build targets and release notes
+  github_release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    name: Create GitHub Release
+    needs: github_build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-unknown-linux-gnu.tar.gz.sha256
+          path: .
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-unknown-linux-musl.tar.gz.sha256
+          path: .
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-apple-darwin.tar.gz.sha256
+          path: .
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-pc-windows-msvc.zip.sha256
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-unknown-linux-gnu.tar.gz
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-unknown-linux-musl.tar.gz
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-apple-darwin.tar.gz
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-pc-windows-msvc.zip
+          path: .
+
+      - name: Install dos2unix to fix windows checksum
+        run: sudo apt-get install -y dos2unix
+
+      - shell: bash
+        run: |
+          # Fix newline issue in windows checksum file and combine all files together
+          dos2unix starship-x86_64-pc-windows-msvc.zip.sha256
+          cat starship-x86_64-pc-windows-msvc.zip.sha256 | xargs > tmp
+          mv tmp starship-x86_64-pc-windows-msvc.zip.sha256
+          cat *.sha256 >> CHECKSUMS.sha256
```

Comment:
Why don't you create the checksums at this stage, so you don't have to run the `download-artifact` action multiple times?

Reply:
![image](https://user-images.githubusercontent.com/6401427/69013769-6f224180-0949-11ea-8777-a652d9ee0b49.png)

that's a good point lol